src/walletSdk/Recovery/index.ts

```diff
@@ -1,32 +1,311 @@
-// TODO - https://stellarorg.atlassian.net/browse/WAL-815?atlOrigin=eyJpIjoiYTE4MTdlZDIwMDU1NGU2YjlhZjQ3MDFlYmJhZWM1NTciLCJwIjoiaiJ9
 import { AxiosInstance } from "axios";
-import { Server } from "stellar-sdk";
+import { Transaction } from "stellar-sdk";
 
 import { Config } from "walletSdk";
-import { Stellar } from "../Horizon";
+import {
+  AccountSigner,
+  AccountThreshold,
+  CommonBuilder,
+  RecoverableAccountInfo,
+  RecoverableWallet,
+  RecoverableWalletConfig,
+  RecoveryAccount,
+  RecoveryAccountInfoMap,
+  RecoveryAccountSigner,
+  RecoveryAuthMap,
+  RecoveryIdentityMap,
+  RecoveryServerKey,
+  RecoveryServerMap,
+} from "walletSdk/Types";
+import { AccountRecover } from "./AccountRecover";
+import { Sep10 } from "../Auth";
+import {
+  DeviceKeyEqualsMasterKeyError,
+  NoAccountAndNoSponsorError,
+  NoAccountSignersError,
+  RecoveryIdentityNotFoundError,
+  ServerRequestFailedError,
+} from "../Exceptions";
+import {
+  AccountKeypair,
+  PublicKeypair,
+  TransactionBuilder,
+  SponsoringBuilder,
+  Stellar,
+} from "../Horizon";
 
 // Let's prevent exporting this constructor type as
 // we should not create this Recovery class directly.
 type RecoveryParams = {
   cfg: Config;
   stellar: Stellar;
   httpClient: AxiosInstance;
-  servers: Server[];
+  servers: RecoveryServerMap;
 };
 
 // Do not create this object directly, use the Wallet class.
-export class Recovery {
+export class Recovery extends AccountRecover {
   private cfg: Config;
-  private stellar: Stellar;
-  private httpClient: AxiosInstance;
-  private servers: Server[];
 
   constructor(params: RecoveryParams) {
     const { cfg, stellar, httpClient, servers } = params;
-
+    super(stellar, httpClient, servers);
     this.cfg = cfg;
-    this.stellar = stellar;
-    this.httpClient = httpClient;
-    this.servers = servers;
+  }
+
+  /**
+   * Register account with recovery servers using
+   * [SEP-30](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0030.md).
+   */
+  private async enrollWithRecoveryServer(
```

Comment:
nit: can we group the private functions together at the bottom of the class? imo it makes the class easier to read when the private funcs are grouped together

Reply:
done